File: backend/app/schemas/report.py
from datetime import datetime
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel, Field


ReportCategory = Literal[
    "POTHOLE",
    "WASTE",
    "STREET_LIGHT",
    "WATER_LEAK",
]


class ReportCreateRequest(BaseModel):
    category: ReportCategory
    description: str | None = Field(default=None, max_length=2000)
    latitude: Decimal = Field(ge=-90, le=90)
    longitude: Decimal = Field(ge=-180, le=180)
    photo_url: str | None = Field(default=None, max_length=500)


class ReportCreateResponse(BaseModel):
    id: str
    category: str
    description: str | None
    latitude: Decimal
    longitude: Decimal
    photo_url: str | None
    status: str
    created_at: datetime

File: frontend/src/types/report.ts
export interface ReportListItem {
  id: string
  category: string
  status: string
  created_at: string
}

export interface ReportDetail {
  id: string
  category: string
  description: string | null
  latitude: number
  longitude: number
  photo_url: string | null
  status: string
  created_at: string
}